src/user-guide/writing-workflows/jinja2.rst

```diff
@@ -120,54 +114,83 @@ variables, loops, math, and logical flags to include or exclude tasks.
+Access to Workflow Files
+------------------------
 
-.. code-block:: cylc
+Your Jinja2 code can see the workflow directory by using
+:ref:`Python code <jinja2.importing_python_modules>`
+that simply reads from the *current working directory*.
 
-   #!Jinja2
-   #...
-   [runtime]
-       [[root]]
-           [[[environment]]]
-               WORKFLOW_OWNER_HOME_DIR_ON_WORKFLOW_HOST = {{environ['HOME']}}
+This will be the source directory if parsing a source workflow, or the run
+directory if parsing an installed workflow.
 
-In addition, the following variables are exported to this environment
-(hence are available in the ``environ`` dict) to provide workflow context:
 
-.. code-block:: sub
+.. _jinja2.workflow_context:
 
-   CYLC_VERBOSE                    # Verbose mode, true or false
-   CYLC_DEBUG                      # Debug mode (even more verbose), true or false
+Workflow Context variables
+--------------------------
 
-   CYLC_WORKFLOW_ID                # Workflow ID
-   CYLC_WORKFLOW_NAME              # Workflow name
-                                   # (the ID with the run name removed)
+Jinja2 CYLC variables available when parsing any workflow (source or installed):
 
-   CYLC_WORKFLOW_LOG_DIR           # Workflow log directory.
-   CYLC_WORKFLOW_RUN_DIR           # Location of the run directory in
-                                   # workflow host, e.g. ~/cylc-run/foo
-   CYLC_WORKFLOW_SHARE_DIR         # Workflow (or task post parsing!)
-                                   # shared directory.
-   CYLC_WORKFLOW_WORK_DIR          # Workflow work directory.
+.. table::
 
-.. warning::
+   ======================    ==============
+   CYLC_VERSION              Version of Cylc parsing the configuration
+   CYLC_WORKFLOW_NAME        Workflow name (source, or run ID minus run name)
+   CYLC_TEMPLATE_VARS        Variables set by '--set' CLI options or plugins
+   ======================    ==============
+
+
+Additional Jinja2 CYLC variables available when parsing an installed workflow:
+
+.. table::
 
-   The environment is read on the workflow host when the configuration is
-   parsed. It is not read at run time by jobs on the job platform.
+   ======================    ==============
+   CYLC_WORKFLOW_ID          Workflow ID
+   ======================    ==============
 
-The following Jinja2 variables are also available (i.e. standalone,
-not in the ``environ`` dict):
 
-``CYLC_VERSION``
-   Version of Cylc used.
+Additional Jinja2 CYLC variables available when the scheduler is parsing an
+installed workflow before starting or restarting a run:
+
+.. table::
+
+   =======================    ==============
+   CYLC_WORKFLOW_RUN_DIR      Workflow run directory
+   CYLC_WORKFLOW_LOG_DIR      Workflow log sub-directory
+   CYLC_WORKFLOW_SHARE_DIR    Workflow share sub-directory
+   CYLC_WORKFLOW_WORK_DIR     Workflow work sub-directory
+   =======================    ==============
```

Comment:
Why separate these from `CYLC_WORKFLOW_ID` above?

Reply:
Because the ID is also available when parsing, but not running, an installed workflow.

Reply:
I've tested out and `CYLC_WORKFLOW_RUN_DIR` is available to `cylc validate` but not the others.

Reply:
```suggestion
Additional Jinja2 CYLC variables available when the scheduler is parsing an
installed workflow at run time:

.. table::

   =======================    ==============
   CYLC_WORKFLOW_LOG_DIR      Workflow log sub-directory
   CYLC_WORKFLOW_SHARE_DIR    Workflow share sub-directory
   CYLC_WORKFLOW_WORK_DIR     Workflow work sub-directory
   =======================    ==============
```

Reply:
> I've tested out and CYLC_WORKFLOW_RUN_DIR is available to cylc validate but not the others.

Oops, yes - for validation of *installed* workflows it is.